### DISCUSSION TITLE
How to setup a proxy with "brunch watch" ?

### DISCUSSION THREAD

**Discussion by Ghuillaume** [brunch/brunch#1870] [Q&A]
### Description

I would like to setup a proxy when my server is running with "brunch watch" command. The aim is to allow my webapp to communicate with a REST API written with Java and running on localhost:8080

Today, my webapp is running on localhost:3333
When I call "GET api/myService" I would like the server to call "GET localhost:8080/api/myService" instead of "GET localhost:3333/api/myService"

How can I do that with brunch ?
Thanks for your help

### `package.json` contents

```json
    "devDependencies": {
        "brunch": "1.7.20",
        "javascript-brunch": "1.7.1",
        "css-brunch": "1.7.0",
        "less-brunch": "1.7.2",
        "auto-reload-brunch": "1.7.8",
        "uglify-js-brunch": "1.7.8",
        "clean-css-brunch": "1.7.2",
        "bower": "1.8.0"
    }
```

### brunch config contents

```coffee
exports.config =
    files:
        javascripts:
            joinTo:
                'js/app.js': /^app/
                'js/vendor.js': /^(bower_components|vendor)/

        stylesheets:
            joinTo:
                'css/app.css': /^(app)/
                'css/vendor.css': [
                    'bower_components/angular-datatables/dist/css/angular-datatables.css',
                    'bower_components/bootstrap/less/bootstrap.less'
                ]

        templates:
            joinTo:
                'css/app.css': /^(app)/

    server:
        port: 3333
        run: yes
```

**Comment by colinbate** [comment]
If you are planning to handle this separately in your production environment, then you can follow the instructions here: http://brunch.io/docs/config#-server- to create a separate `brunch-server.js` file which you can use to serve up your `public` directory as well as proxy any other non-static requests through to your Java API. You might be able to use something like this to help: https://www.npmjs.com/package/static-proxy

**Comment by Ghuillaume** [comment]
Thanks for the tip, but it doesn't seem to work.
I'm able to serve my static files, and I can see in the logs that my API is proxified. But I have a HTTP400 error in my brower.


Do you have any idea why this doesn't work ?


Logs : 
```log
30 May 10:51:10 - info: starting custom server
static-proxy listening at http://0.0.0.0:3333
30 May 10:51:16 - warn: custom server taking a long time to start
30 May 10:51:16 - warn: **don't forget to invoke callback()**
Requesting url >>  http://localhost:8080\myApi\myService
host: localhost:8080/myApi/
```


If I call http://localhost:8080\myApi\myService by myself I have a correct answer.
In my webapp, I have an error on this : http://localhost:3333/api/myService


My brunch-server.js file is : 
```js
staticProxy = require('static-proxy');

exports.startServer = (port, path, callback) => {
    staticProxy('localhost:8080/myApi/', port, 'http', [path], true, transform);
};
```

My config.coffee file is :
```coffee
exports.config =
    server:
        path: 'brunch-server.js'
        port: 3333
        run: yes
```

**Comment by shvaikalesh** [comment]
We actually don't support older Brunches. Try migrating to `^2`, that should be a piece of cake.

**Comment by colinbate** [comment]
Oh yea, I didn't even notice the Brunch 1 reference. That would be the first thing to try. Also if you aren't invoking `callback` in your server function, then don't include it as a parameter.
